src/safeposix/syscalls/fs_calls.rs

```diff
@@ -441,25 +475,39 @@ impl Cage {
                 let newinodenum = FS_METADATA
                     .nextinode
                     .fetch_add(1, interface::RustAtomicOrdering::Relaxed); //fetch_add returns the previous value, which is the inode number we want
-                if let Inode::Dir(ref mut parentdir) =
-                    *(FS_METADATA.inodetable.get_mut(&pardirinode).unwrap())
+                
+                // Insert a reference to the file in the parent directory and update the inode attributes
+                // Fetch the inode of the parent directory and only proceed when its type is directory.
+                if let Inode::Dir(ref mut parentdir) = *(FS_METADATA.inodetable.get_mut(&pardirinode).unwrap())
                 {
                     parentdir
                         .filename_to_inode_dict
                         .insert(filename, newinodenum);
                     parentdir.linkcount += 1;
-                } //insert a reference to the file in the parent directory
+                    parentdir.ctime = time; // Here, update the ctime and mtime for the parent directory as well
```

Comment:
Makes sense to me.  In general mentioning this in your issue / PR text is preferred.

Reply:
Sure.